src/main/java/org/alfresco/repo/security/authentication/InMemoryTicketComponentImpl.java

```diff
@@ -491,10 +522,34 @@ private Ticket(ExpiryMode expires, Date expiryDate, String userName, Duration va
             this.ticketId = ticketId;
             if (logger.isTraceEnabled())
             {
-                logger.trace("Creating new ticket for user:" + AuthenticationUtil.maskUsername(userName));
+                logger.trace("Creating (cloning) new ticket for user: " + AuthenticationUtil.maskUsername(userName));
             }
         }
-        
+
+        private static void checkValidTicketParameters(ExpiryMode expires, Date expiryDate, String userName, Duration validDuration)
+        {
+            checkParameter(expires, "expires mode");
+            checkParameter(userName, "userName");
+            checkParameter(validDuration, "validDuration");
+            if (!ExpiryMode.DO_NOT_EXPIRE.equals(expires))
+            {
+                checkParameter(expiryDate, "expiryDate");
+            }
+        }
+
+        private static String getExpireDateAsString(Date expiryDate)
+        {
+            return (expiryDate == null) ? "" : expiryDate.toString();
+        }
+
+        private static void checkParameter(Object parameter, String name)
```

Comment:
There is a similar method in org.alfresco.util.ParameterCheck.mandatory()

Reply:
Good find. I did change that. Thanks